src/AppController.m

```diff
@@ -333,14 +336,18 @@ -(void)installSleepHandler
  * Install a handler to notify of changes in the scripts folder.
  * The handler is a code block which triggers a refresh of the scripts menu
  */
--(void)installScriptsFolderWatcher
-{
-	NSURL * path = [NSURL fileURLWithPath:[Preferences standardPreferences].scriptsFolder];
-	_events = [[CDEvents alloc] initWithURLs:@[path]
-                                       block:^(CDEvents *watcher, CDEvent *event) {
-										   // triggers a refresh of the scripts.menu
-                                           [self initScriptsMenu];
-                                       }];
+- (void)installScriptsFolderWatcher {
+    NSURL *path = [NSURL fileURLWithPath:Preferences.standardPreferences.scriptsFolder];
+    self.directoryMonitor = [[DirectoryMonitor alloc] initWithDirectories:@[path]];
+
+    NSError *error = nil;
+    [self.directoryMonitor startAndReturnError:&error eventHandler:^{
+        typeof(self) __weak weakSelf = self;
```

Comment:
Nope. I did not know it, I looked this up and the first example I found declared it inside the closure. I assumed that this was correct, because Swift does it like that too.

Apple seems to declare it outside of the closure too: https://developer.apple.com/library/content/documentation/Cocoa/Conceptual/ProgrammingWithObjectiveC/WorkingwithBlocks/WorkingwithBlocks.html#//apple_ref/doc/uid/TP40011210-CH8-SW16.